common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthRequestProvider.kt

```diff
@@ -310,7 +311,11 @@ class NativeAuthRequestProvider(private val config: NativeAuthOAuth2Configuratio
     //region helpers
     private fun getRequestHeaders(correlationId: String): Map<String, String?> {
         val headers: MutableMap<String, String?> = TreeMap()
-        headers[AuthenticationConstants.AAD.CLIENT_REQUEST_ID] = correlationId
+
+        if (correlationId != "UNSET") {
+            headers[AuthenticationConstants.AAD.CLIENT_REQUEST_ID] = correlationId
+        }
+
```

Comment:
What about correlationId == "UNSET"? 
` val privateCorrelationId = if (correlationId == "UNSET") { UUID.randomUUID().toString() } else { correlationId }` only for AccountState `acquireTokenSilentParameters`. Do we have other cases that may have UNSET?

Reply:
If it's equal to UNSET, we don't want to attach that UNSET value to the header, because then the server will return that back to us. We want to send a nil header so that the server will generate a correlation ID for us if it isn't provided. This is the block of code that configures the request headers for every request so it's a good place to check it here.

In terms of scenarios where it's possible the value is "UNSET", the example you gave is one. Additionally, for every flow that's initiated (sign in, sign up, reset password), the correlation ID will also be initialised as "UNSET". The check is added here as a 'catch-all' for those flows.

Reply:
> Additionally, for every flow that's initiated (sign in, sign up, reset password), the correlation ID will also be initialised as "UNSET".

@rmccahill why is that the case? In practise the correlation ID should never be set to `UNSET`. At the start of the flow, i.e. in `CommandParametersAdapter`, the ID is generated for `signIn()`, `signUp()` and `resetPassword()` (i.e. methods that start the flow).

Reply:
You're correct - I misread the underlying code, the DiagnosticContext defaults the correlation ID to be "UNSET", but the CommandDispatcher updates this when it runs a command. So the AccountState example should be the only case where this happens (that we know of at least)